packages/babel-parser/src/plugins/typescript.js

```diff
@@ -1539,11 +1539,13 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       // export default interface allowed in:
       // https://github.com/Microsoft/TypeScript/pull/16040
       if (this.state.value === "interface") {
-        return this.tsParseDeclaration(
+        const result = this.tsParseDeclaration(
           this.startNode(),
           this.state.value,
           true,
         );
+
+        if (result) return result;
```

Comment:
This changes the behavior of Babylon. Is it intended? If so, can you extract it to a separate PR and add a test?

Reply:
Yeah, new errors detected from latest Flow, basically `tsParseDeclaration` can return `undefined` and the typing of `parseExportDefaultExpression` doesn't allow it.

Reply:
Nah it's ok